app/screens/planner/planner-screen-header.tsx

```diff
@@ -45,54 +47,75 @@ export const PlannerScreenHeader = observer(function PlannerScreenHeader() {
+              onPress={() => {
+                // @ts-expect-error
+                navigation.navigate("activeRideStack", {
+                  screen: "activeRide",
+                  params: { routeItem: ride.route, originId: ride.originId, destinationId: ride.destinationId },
+                })
 
-              analytics().logEvent("open_live_ride_modal_pressed")
-            }}
-          >
-            {Platform.OS === "ios" && <Image source={TRAIN_ICON} style={LIVE_BUTTON_IMAGE} />}
-            <Text style={{ color: "white", fontWeight: "500" }} tx="ride.live" />
-          </Chip>
-        )}
+                analytics().logEvent("open_live_ride_modal_pressed")
+              }}
+            >
+              {Platform.OS === "ios" && <Image source={TRAIN_ICON} style={LIVE_BUTTON_IMAGE} />}
+              <Text style={{ color: "white", fontWeight: "500" }} tx="ride.live" />
+            </Chip>
+          )}
 
-        {displayNewBadge && (
-          <Chip color="primary" onPress={() => navigation.navigate("liveAnnouncementStack")}>
-            <Image source={SPARKLES_ICON} style={{ height: 16, width: 16, marginEnd: spacing[2], tintColor: "white" }} />
-            <Text style={{ color: "white", fontWeight: "500" }} tx="common.new" />
-          </Chip>
-        )}
+          {displayNewBadge && (
+            <Chip color="primary" onPress={() => navigation.navigate("liveAnnouncementStack")}>
+              <Image source={SPARKLES_ICON} style={{ height: 16, width: 16, marginEnd: spacing[2], tintColor: "white" }} />
+              <Text style={{ color: "white", fontWeight: "500" }} tx="common.new" />
+            </Chip>
+          )}
+        </View>
+        <TouchableOpacity
+          onPress={() => navigation.navigate("announcementsStack")}
+          activeOpacity={0.8}
+          accessibilityLabel={translate("routes.updates")}
+        >
+          <Image source={UPDATES_ICON} style={SETTINGS_ICON_IMAGE} />
+        </TouchableOpacity>
+        <TouchableOpacity onPress={() => navigation.navigate("settingsStack")} activeOpacity={0.8} accessibilityLabel="הגדרות">
+          <Image source={SETTINGS_ICON} style={SETTINGS_ICON_IMAGE} />
+        </TouchableOpacity>
       </View>
-      <TouchableOpacity
-        onPress={() => navigation.navigate("announcementsStack")}
-        activeOpacity={0.8}
-        accessibilityLabel={translate("routes.updates")}
-      >
-        <Image source={UPDATES_ICON} style={SETTINGS_ICON_IMAGE} />
-      </TouchableOpacity>
-      <TouchableOpacity onPress={() => navigation.navigate("settingsStack")} activeOpacity={0.8} accessibilityLabel="הגדרות">
-        <Image source={SETTINGS_ICON} style={SETTINGS_ICON_IMAGE} />
-      </TouchableOpacity>
-    </View>
+
+      {showUrgentBar && (
+        <View style={{ position: "absolute", top: 0, left: 16 }}>
+          <ImportantAnnouncementBar title={`${popupMessages[0].messageBody.slice(0, 41)}...`} />
+        </View>
+      )}
```

Comment:
It shouldn't hide the other buttons in the header IMO, and can't styles truncate automatically?